### ISSUE TITLE
[epic] Bring in maven plugin, archetype and java codegen

### ISSUE THREAD

**Issue by octonato** [lightbend/akkaserverless-java-sdk#27]
As discussed, we plan to merge into this project the maven plugin, the maven archetypes and the java codegen. 

The execution plan is as follow: 

## Phase 1:

- [x] **move**: bring `maven-java` under `java-sdk` (preserving commit history) #28 
- [x] **build**: must be integrated in CircleCI (similar to what we are doing to the samples already) https://github.com/lightbend/akkaserverless-java-sdk/pull/36
- [x] **versioning**: we want the maven plugin and archetypes to have the same version of the java-sdk. https://github.com/lightbend/akkaserverless-java-sdk/pull/65
- [x] **publishing**: archetypes and plugin must be published under a new name because of version misalignment https://github.com/lightbend/akkaserverless-java-sdk/pull/45 and https://github.com/lightbend/akkaserverless-java-sdk/pull/65
- [x] **docs**: documentation must be adapted since users won't need to change anything in their `settings.xml` file and we will have a new artifact name #96 
- [x] **github issues**: move all open issues from `maven-java` to `java-sdk`

## Phase 2:

First we need to split codegen into codegen-core, codegen-java and codegen-js (preserving commit history) See https://github.com/lightbend/akkaserverless-codegen/issues/147

- [x] move `codegen-java` under `java-sdk` (preserving commit history)

Please, feel free to add here any other point of action that you believe we will need.

**Comment by ennru** [comment]
@octonato Common believe is that everything is covered now. It would be good if you'd check and try a release.

**Comment by octonato** [comment]
Yep, I think we have everything.

**Comment by octonato** [comment]
For the record, the publishing of artifacts didn't work. The reason is that we call `publishM2` a few times in the build and then we get a message that artifact is already published (in local).

That brings us back to the cache issue. We should call `publishM2` only once in the build, but we need to get to understand why the m2 cache is not working as expected.

**Comment by octonato** [comment]
After a few build tweaks and burned release numbers we have v0.7.0-beta.15 released including the codegen, the maven plugin, the archetypes and the SDK all together. 

We can close this issue! 🎉
